homeassistant/components/ihc/light.py

```diff
@@ -79,14 +86,20 @@ def turn_on(self, **kwargs) -> None:
             self.ihc_controller.set_runtime_value_int(
                 self.ihc_id, int(brightness * 100 / 255))
         else:
-            self.ihc_controller.set_runtime_value_bool(self.ihc_id, True)
+            if self._ihc_on_id:
+                await pulse(self.ihc_controller, self._ihc_on_id)
+            else:
+                self.ihc_controller.set_runtime_value_bool(self.ihc_id, True)
```

Comment:
Is this async safe? Ie no I/O or blocking being done?

Reply:
No it does I/O, so I changed these to be called with hass.async_add_executor_job